txgossip/pushpull.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txgossip
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/network/p2p/gossip"
+	"github.com/ava-labs/libevm/core/txpool"
+	"github.com/ava-labs/libevm/core/types"
+)
+
+// Add is a wrapper around [txpool.TxPool.Add], exposed to accept transactions
+// over [gossip]. Populating the [Set] via this method does *not* result in new
+// transactions being push-gossiped, but they will be included in pull-gossip
+// responses.
+func (s *Set) Add(tx Transaction) error {
+	errs := s.Pool.Add([]*types.Transaction{tx.Transaction}, false, false)
+	for i, err := range errs {
+		if errors.Is(err, txpool.ErrAlreadyKnown) {
+			errs[i] = nil
+		}
```

Comment:
Because we are replacing this error with `nil`, we are going to end up calling `s.bloom.Add` even if the tx was already known. This can result in us resetting our bloom filter more frequently than expected.